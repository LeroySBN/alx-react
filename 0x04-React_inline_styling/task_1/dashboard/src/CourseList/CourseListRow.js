import React from 'react';
import PropTypes from 'prop-types';

export default function CourseListRow({ isHeader, textFirstCell, textSecondCell }) {
  const tableHeaderStyle = {
    backgroundColor: "#deb5b545",
  };
  const tableRowStyle = {
    backgroundColor: "#f5f5f5ab",
  };
  return (
    <tr>
      {isHeader ? (
        textSecondCell === null ? (
          <th colSpan={2}>{textFirstCell}</th>
        ) : (
          <>
            <th style={tableHeaderStyle}>{textFirstCell}</th>
            <th style={tableHeaderStyle}>{textSecondCell}</th>
          </>
        )
      ) : (
        <>
          <td style={tableRowStyle}>{textFirstCell}</td>
          <td style={tableRowStyle}>{textSecondCell}</td>
        </>
      )}
    </tr>
  );
}

CourseListRow.propTypes = {
    isHeader: PropTypes.bool,
    textFirstCell: PropTypes.string.isRequired,
    textSecondCell: PropTypes.oneOfType([
      PropTypes.number,
      PropTypes.string,
    ]),
}

CourseListRow.defaultProps = {
    isHeader: false,
    textSecondCell: null,
}
