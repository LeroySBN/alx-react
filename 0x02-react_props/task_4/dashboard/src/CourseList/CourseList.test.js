import React from 'react';
import { shallow } from 'enzyme';
import CourseList from './CourseList';

describe('CourseList component testing', () => {
  it('CourseList renders without crashing', () => {
    const component = shallow(<CourseList />);
    expect(component).toMatchSnapshot();
  });

  it('renders a table', () => {
    const wrapper = shallow(<CourseList />);
    expect(wrapper.find('table')).toHaveLength(1);
    expect(wrapper.find('CourseListRow')).toHaveLength(5);
  });
});