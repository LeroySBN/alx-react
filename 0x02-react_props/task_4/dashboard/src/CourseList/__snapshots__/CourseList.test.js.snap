// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseList component testing CourseList renders without crashing 1`] = `ShallowWrapper {}`;
